sdk/core/core/inc/az_version.h

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file az_version.h
+ *
+ * @brief Provides version information
+ *
+ * NOTE: You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ */
+
+#ifndef _az_VERSION_H
+#define _az_VERSION_H
+
+int az_core_version_get_major();
+int az_core_version_get_minor();
```

Comment:
We should probably add a get_version_string API that returns the exact version string composing the various semver parts (effectively the string that recreates the version.txt file).

Reply:
Functions are removed, version.txt will be removed as well.  az_version.h will be the single source of truth